### PR TITLE
Introduce regex matching in Consumer#subscribe

### LINKED ISSUE #299: Feature Request: subscribe to multiple topics by regex
This feature is [supported](https://kafka.apache.org/0101/javadoc/org/apache/kafka/clients/consumer/Consumer.html#subscribe(java.util.regex.Pattern,%20org.apache.kafka.clients.consumer.ConsumerRebalanceListener)) by the canonical Java client and would be useful for my company’s architecture.

Thanks!

### PR DESCRIPTION
**Pull request by rkrage**
Resolves #299 

This PR certainly isn't ready to merge yet. A few notes / questions:

- This doesn't auto-discover new topics after the consumer starts - not sure what that implementation would look like
- Do you have any ideas on how to avoid the code duplication introduced in the `cluster_topics` method
- What kind of tests would be worthwhile for this?

### CODE REVIEW

File: lib/kafka/consumer.rb [hunk 1]
```diff
@@ -103,6 +103,21 @@ def subscribe(topic, default_offset: nil, start_from_beginning: true, max_bytes_
       nil
     end
 
+    def subscribe_by_regex(topic_regex, default_offset: nil, start_from_beginning: true, max_bytes_per_partition: 1048576)
```
Comment (dasch) [inline lib/kafka/consumer.rb]: Could we call this e.g. `subscribe_matching` instead?

Comment (dasch) [reply lib/kafka/consumer.rb]: Or maybe just merge this into `subscribe`? I actually think it's better to just check if the user passes a regex there. Then existing integrations can just start passing regexes.

File: lib/kafka/consumer.rb [hunk 2]
```diff
@@ -103,6 +103,21 @@ def subscribe(topic, default_offset: nil, start_from_beginning: true, max_bytes_
...
+      cluster_topics.each do |topic|
+        next unless topic =~ topic_regex
```
Comment (dasch) [inline lib/kafka/consumer.rb]: `cluster_topics.select {|topic| topic =~ topic_regex }.each do |topic|

**Comment by rkrage** [comment]
If this looks alright, I'll add some functional tests and update the inline documentation

**Comment by mensfeld** [comment]
I did ;) LGTM!

**Comment by dasch** [comment]
Thanks!

**Pull request merged**